packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -38,9 +39,55 @@ export class GitQuickOpenService {
         @inject(Git) protected readonly git: Git,
         @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider,
         @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
-        @inject(MessageService) protected readonly messageService: MessageService
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
     ) { }
 
+    async clone(url?: string, folder?: string, branch?: string): Promise<void> {
+        if (!folder) {
+            const roots = await this.workspaceService.roots;
+            folder = roots[0].uri;
+        }
+
+        const __folder = folder;
+        if (url) {
+            return this.git.clone(url, { localUri: this.buildDefaultProjectPath(folder, url), branch: branch }).then(() => { });
+        }
+
+        const __this = this;
+        const cloneRepoModel: QuickOpenModel = {
+            onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+                const dynamicItems: QuickOpenItem[] = [];
+                const suffix = "Press 'Enter' to confirm or 'Escape' to cancel.";
+                if (lookFor === undefined || lookFor.length === 0) {
+                    dynamicItems.push(new SingleStringInputOpenItem(`Please provide a Git repository URI. ${suffix}`, () => { }, () => false));
+                } else {
+                    dynamicItems.push(new SingleStringInputOpenItem(
+                        `Clone the Git Repository: ${lookFor}. ${suffix}`,
+                        async () => {
+                            try {
+                                await __this.git.clone(lookFor, { localUri: __this.buildDefaultProjectPath(__folder, lookFor) });
+                            } catch (error) {
+                                __this.gitErrorHandler.handleError(error);
+                            }
+                        }
+                    ));
+                }
+                acceptor(dynamicItems);
+            }
+        };
+        this.quickOpenService.open(cloneRepoModel, this.getOptions('Git Repository URI:', false));
+    }
+
+    buildDefaultProjectPath(projectRoot: string, gitURI: string): string {
```

Comment:
- Should have `private` visibility.
 - Please rename `projectRoot` to `projectRootUri` or something, so that it reflects, it is not an FS path but a URI string. Thanks!

Reply:
it could be both actually. So renamed to `folderPath`.